lib/ace/ext/papeeria/papeeria_latex_highlight_rules.coffee

```diff
@@ -30,6 +32,14 @@ define((require, exports, module) ->
           stack.push(pushedState)
         return pushedState
 
+    pushStateCheckout = (pushedState, to) ->
```

Comment:
Any comment on what's going on here? 

Overall, this code needs a dozen of tests.